### ISSUE TITLE
Sparknlp returning different embedding for manual spark dataframe vs reading from file spark dataframe

### ISSUE THREAD

**Issue by rishabhindoria** [JohnSnowLabs/spark-nlp#14182]
### Is there an existing issue for this?

- [X] I have searched the existing issues and did not find a match.

### What are you working on?

I am trying out MPNetEmbeddings to get embeddings for names 

### Current Behavior

Hi team, I m trying out  MPNetEmbeddings in Pyspark shell on command line connected to a spark cluster to get embeddings for names but it seems the library is giving random output? Here is example - when I create spark dataframe with manual input 3 names - it returns correct embedding but when I give it a spark csv dataframe with just one column having 3 names it gives totally different embedding for the same name "sara"? Can someone please help me understand what is going on? Thank you 

### Expected Behavior

Consistent embedding output for same name regardless of manually created dataframe or dataframe created from reading from a file

### Steps To Reproduce

Here is fully reproducible code for anyone:
working gives correct embeddings matching with huggingface exactly!

pyspark --master yarn --conf spark.shuffle.service.enabled=true --conf spark.hadoop.hive.cli.print.header=true --conf spark.hadoop.hive.exec.dynamic.partition.mode=nonstrict --conf spark.ui.port=0 --conf spark.serializer=org.apache.spark.serializer.KryoSerializer --packages  com.johnsnowlabs.nlp:spark-nlp_2.12:5.2.3 --conf spark.dynamicAllocation.enabled=false --conf spark.kryoserializer.buffer.max=2000m

from pyspark.sql.functions import *
import sparknlp
from sparknlp.base import *
from sparknlp.annotator import *
from pyspark.ml import Pipeline
documentAssembler = DocumentAssembler().setInputCol("text").setOutputCol("document")
embeddings = MPNetEmbeddings.pretrained().setInputCols(["document"]).setOutputCol("mpnet_embeddings")
embeddingsFinisher = EmbeddingsFinisher().setInputCols(["mpnet_embeddings"]).setOutputCols("finished_embeddings").setOutputAsVector(True)
pipeline = Pipeline().setStages([documentAssembler, embeddings, embeddingsFinisher])
empty_df = spark.createDataFrame([['']]).toDF("text")
data = spark.createDataFrame([["sara","john","rishabh"]]).toDF("text")
result = pipeline.fit(empty_df).transform(data).withColumn("finished_embeddings", explode("finished_embeddings")).drop("mpnet_embeddings","document")
result.where("text=='sara'").show(1,100)
+----+----------------------------------------------------------------------------------------------------+
|text|                                                                                 finished_embeddings|
+----+----------------------------------------------------------------------------------------------------+
|sara|[-0.020321492105722427,0.03794000670313835,-0.0016927680699154735,-0.034501343965530396,0.0137658...|
+----+----------------------------------------------------------------------------------------------------+

not working and giving totally different embeddings just 1 change is reading from a csv of just 1 column called "text" with the same 3 names ie "sara", "mara" and "rishabh"?

from pyspark.sql.functions import *
import sparknlp
from sparknlp.base import *
from sparknlp.annotator import *
from pyspark.ml import Pipeline
documentAssembler = DocumentAssembler().setInputCol("text").setOutputCol("document")
embeddings = MPNetEmbeddings.pretrained().setInputCols(["document"]).setOutputCol("mpnet_embeddings")
embeddingsFinisher = EmbeddingsFinisher().setInputCols(["mpnet_embeddings"]).setOutputCols("finished_embeddings").setOutputAsVector(True)
pipeline = Pipeline().setStages([documentAssembler, embeddings, embeddingsFinisher])
empty_df = spark.createDataFrame([['']]).toDF("text")
data = spark.read.option("header","true").csv("names.csv")
result = pipeline.fit(empty_df).transform(data).withColumn("finished_embeddings", explode("finished_embeddings")).drop("mpnet_embeddings","document")
result.where("text=='sara'").show(1,100)

+----+----------------------------------------------------------------------------------------------------+                                                                               
|text|                                                                                 finished_embeddings|
+----+----------------------------------------------------------------------------------------------------+
|sara|[0.017252618446946144,0.006142440717667341,-0.0015354102943092585,-0.054641805589199066,0.0220024...|
+----+----------------------------------------------------------------------------------------------------+

names.csv looks like:
+-------+               
|text   |
+-------+
|mara   |
|sara   |
|rishabh|
+-------+

### Spark NLP version and Apache Spark

sparknlp version: 5.2.3
Python version 3.10.13 
spark version: 3.3.2

### Type of Spark Application

spark-shell, Python Application

**Comment by maziyarpanahi** [comment]
Hi @rishabhindoria 

I think it's because in one of those it batches the inputs. Could you please add `.setBatchSize(1)` to the MPNet annotator and try again?

**Comment by rishabhindoria** [comment]
Thank you @maziyarpanahi but then it won't be able to scale right if the batch size itself is 1 record at a time? Need to apply this to millions of names
What is the default batch size?

**Comment by maziyarpanahi** [comment]
I think it’s 4 or 8 but it’s just a default number. 
It’s the same with any other library, if you batch a different group of sequences for inference, the output will change due to padding. (Doesn’t mean the quality degrades, it’s just different)

**Comment by rishabhindoria** [comment]
thank you @maziyarpanahi it works with setting batch size=1 but again how to scale it to hundreds of millions of names? do we need to set 1 cpu core per executor and lots of such executors like this so as to parallelize?
--conf spark.executor.instances=2000 --conf spark.executor.cores=1
any guidance from other projects/companies using that in production or youtube video on that would really help

**Comment by maziyarpanahi** [comment]
You are welcome. As I mentioned those discrepancies are minor and unavoidable. The moment you batch any inputs you must either pad or truncate. This results in small precision changes as you noticed.
I suggest tuning the batchSize for performance and scale specially on GPUs and not too worried about that small changes. (We have trained high-accurate models without any issues)
